File: base.database/src/main/java/alvin/base/database/dbflow/DBFlowModule.java
package alvin.base.database.dbflow;

import alvin.base.database.dbflow.presenters.DBFlowPresenter;
import alvin.base.database.dbflow.views.DBFlowActivity;
import dagger.Binds;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public interface DBFlowModule {

    @ContributesAndroidInjector(modules = ActivityModule.class)
    DBFlowActivity dbFlowActivity();

    @Module
    interface ActivityModule {
        @Binds
        DBFlowContracts.View view(DBFlowActivity activity);

        @Binds
        DBFlowContracts.Presenter presenter(DBFlowPresenter presenter);
    }
}

File: base.kotlin/src/main/java/alvin/base/kotlin/dbflow/DBFlowModule.kt
@file:Suppress("unused")

package alvin.base.kotlin.dbflow

import alvin.base.kotlin.dbflow.presenters.DBFlowPresenter
import alvin.base.kotlin.dbflow.views.DBFlowActivity
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
interface DBFlowModule {

    @ContributesAndroidInjector(modules = [ViewModule::class])
    fun dBFlowActivity(): DBFlowActivity

    @Module
    abstract class ViewModule {

        @Binds
        abstract fun view(activity: DBFlowActivity): DBFlowContract.View

        @Binds
        abstract fun presenter(presenter: DBFlowPresenter): DBFlowContract.Presenter
    }
}
